### PR TITLE
[TO] Added resource mapping use case

### PR DESCRIPTION
**Pull request by hussnain612**

### CODE REVIEW

**Comment by stakater-user** [comment]
@hussnain612 Image is available for testing. `docker pull docker-stakater-nexus.apps.prod.clusters.stakater.io/stakater/stakaterclouddocs:SNAPSHOT-PR-53-b4d39e28`

File: docs/content/sre/tenant-operator/customresources.md [hunk 1]
```diff
@@ -227,11 +227,24 @@ resources:
...
+    configMaps:
+      - name: configmap-c1
+        namespace: namespace-n2
 ```
 
-Templates are used to initialize Namespaces and share common resources across namespaces (e.g. secrets).
+Templates are used to initialize Namespaces, share common resources across namespaces, and map secrets and configMaps from one namespace to other namespaces.
 
-* They either contains one or more Kubernetes manifests or alternatively a Helm chart.
+* They either contains one or more Kubernetes manifests, reference to some secrets and configMaps or alternatively a Helm chart.
```
Comment (MahnoorAsghar) [inline docs/content/sre/tenant-operator/customresources.md]: They either **contain** one or more Kubernetes manifests, a reference to a secret or configMap, or a Helm chart. -- grammar + reads better

File: docs/content/sre/tenant-operator/usecases/template.md [hunk 2]
```diff
@@ -38,6 +38,24 @@ resources:
...
+  resourceMappings:
+    secrets:
+      - name: docker-secret
+        namespace: bluesky-build
+    configMaps:
+      - name: tronador-configMap
+        namespace: stakater-tronador
+```
+
+**Note:** Resource mapping could only be used via TGI to map resources within tenant namespaces or to some other tenant's namespace.
```
Comment (MahnoorAsghar) [inline docs/content/sre/tenant-operator/usecases/template.md]: Resource mapping **can** only be used via TGI to map resources within tenant namespaces, or to some other tenant's namespace.

File: docs/content/sre/tenant-operator/customresources.md [hunk 3]
```diff
@@ -245,6 +258,10 @@ Also you can define custom variables in `Template` and `TemplateInstance` . The
   <summary> Helm Chart Templates</summary>
   <p>Instead of manifests, a Template can specify a Helm chart that will be installed (using Helm template) when the Template is being instantiated.</p>
 </details>
+<details open>
+  <summary>Resource Mapping Templates</summary>
+  <p>The templates can be specified with secrets and configMaps that can be mapped from one tenant's namespace to some other tenant's namespace or within a tenant's namespace.</p>
```
Comment (MahnoorAsghar) [inline docs/content/sre/tenant-operator/customresources.md]: A template can be used to map secrets and configMaps from one tenant's namespace to another tenant's namespace, or within a tenant's namespace.

Comment (MahnoorAsghar) [reply docs/content/sre/tenant-operator/customresources.md]: The rest of the points are using a more active voice, so this point should too. It also sounds clearer :)

**Comment by stakater-user** [comment]
@hussnain612 Image is available for testing. `docker pull docker-stakater-nexus.apps.prod.clusters.stakater.io/stakater/stakaterclouddocs:SNAPSHOT-PR-53-c412114e`

File: docs/content/sre/tenant-operator/customresources.md [hunk 4]
```diff
@@ -227,11 +227,24 @@ resources:
...
+    secrets:
+      - name: secret-s1
+        namespace: namespace-n1
+    configMaps:
+      - name: configmap-c1
+        namespace: namespace-n2
 ```
 
-Templates are used to initialize Namespaces and share common resources across namespaces (e.g. secrets).
+Templates are used to initialize Namespaces, share common resources across namespaces, and map secrets and configMaps from one namespace to other namespaces.
```
Comment (abdulhaseeb2) [inline docs/content/sre/tenant-operator/customresources.md]: 'and map secret/configMaps from one namespace to other namespaces.'
^ Update it to this if this makes more sense.

File: docs/content/sre/tenant-operator/customresources.md [hunk 5]
```diff
@@ -227,11 +227,24 @@ resources:
...
 
-* They either contains one or more Kubernetes manifests or alternatively a Helm chart.
+* They either contain one or more Kubernetes manifests, a reference to secrets or configMaps, or a Helm chart.
```
Comment (abdulhaseeb2) [inline docs/content/sre/tenant-operator/customresources.md]: `a reference to secrets/configMaps, or a Helm chart.`
same here

**Review by abdulhaseeb2 — CHANGES_REQUESTED** [review]

**Comment by stakater-user** [comment]
@hussnain612 Image is available for testing. `docker pull docker-stakater-nexus.apps.prod.clusters.stakater.io/stakater/stakaterclouddocs:SNAPSHOT-PR-53-c5a8dc2e`

**Comment by stakater-user** [comment]
@LilShah Image is available for testing. `docker pull docker-stakater-nexus.apps.prod.clusters.stakater.io/stakater/stakaterclouddocs:SNAPSHOT-PR-53-1eed4068`

**Comment by stakater-user** [comment]
@LilShah Image is available for testing. `docker pull docker-stakater-nexus.apps.prod.clusters.stakater.io/stakater/stakaterclouddocs:SNAPSHOT-PR-53-2f886729`

**Review by abdulhaseeb2 — APPROVED** [review]

**Comment by stakater-user** [comment]
@hussnain612 Image is available for testing. `docker pull docker-stakater-nexus.apps.prod.clusters.stakater.io/stakater/stakaterclouddocs:v0.0.33`

**Pull request merged**